src/pheromones/src/GetNeighborsListResponseDTO.cpp

```diff
@@ -0,0 +1,47 @@
+#include "GetNeighborsListResponseDTO.h"
+
+GetNeighborsListResponseDTO::GetNeighborsListResponseDTO(const GetNeighborsListResponse& resp) {
+    setNeighbors(resp.neighbors, resp.neighbors_count);
+}
+
+GetNeighborsListResponseDTO::GetNeighborsListResponseDTO(uint16_t* neighbors,
+                                                         uint16_t neighborsLength) {
+    setNeighbors(neighbors, neighborsLength);
+}
+
+const std::array<uint16_t, GetNeighborsListResponseDTO::NEIGHBORS_MAX_SIZE>&
+GetNeighborsListResponseDTO::getNeighbors() const {
+    return m_neighbors;
+}
+
+std::array<uint16_t, GetNeighborsListResponseDTO::NEIGHBORS_MAX_SIZE>& GetNeighborsListResponseDTO::
+    getRawNeighbors() {
+    return m_neighbors;
+}
+
+uint16_t GetNeighborsListResponseDTO::getNeighborsLength() const { return m_neighborsLength; }
+
+bool GetNeighborsListResponseDTO::setRawNeighborsLength(uint16_t neighborsLength) {
+    m_neighborsLength = neighborsLength < m_neighbors.size() ? neighborsLength : m_neighbors.size();
+    return m_neighborsLength == neighborsLength;
+}
+
+void GetNeighborsListResponseDTO::setNeighbors(const uint16_t* neighbors,
+                                               uint16_t neighborsLength) {
+    setRawNeighborsLength(neighborsLength);
+    memcpy(m_neighbors.data(), neighbors, m_neighborsLength);
+}
+
+void GetNeighborsListResponseDTO::setNeighbors(const uint32_t* neighbors,
```

Comment:
what differs this method from the upper one?

Reply:
Good catch, not necessary, wil remove. was during dev

Reply:
Good catch, not necessary, will remove. Forgot to remove during cleanup